lib/cloud_controller/metrics/prometheus_updater.rb

```diff
@@ -62,70 +78,46 @@ def update_thread_info(thread_info)
       update_gauge_metric(:cc_thread_info_event_machine_resultqueue_num_waiting, thread_info[:event_machine][:resultqueue][:num_waiting], 'EventMachine requests waiting in queue')
     end
 
-    def update_failed_job_count(failed_jobs_by_queue, total)
+    def update_failed_job_count(failed_jobs_by_queue)
       failed_jobs_by_queue.each do |key, value|
-        metric_key = :"cc_failed_job_count_#{key.to_s.underscore}"
-        update_gauge_metric(metric_key, value, "Failed jobs for worker #{key}")
+        update_gauge_metric(:cc_failed_jobs_total, value, "Failed jobs for worker #{key}", labels: { queue: key.to_s.underscore })
       end
-
-      update_gauge_metric(:cc_failed_job_count_total, total, 'Total failed jobs')
     end
 
     def update_vitals(vitals)
       vitals.each do |key, value|
+        next if key.to_s.underscore == 'cpu'
+
         metric_key = :"cc_vitals_#{key.to_s.underscore}"
```

Comment:
I'm wondering how these metrics would look like for Puma with multiple workers (processes)... If we want to emit memory metrics on our own, the `periodic_updater` needs to run for every process. For the other metrics, that are fetched from the database, we only need one `periodic_updater` for an entire VM.

Reply:
We could try to keep the memory metric as a sum of all Puma subprocesses...

Reply:
The code changes in this PR do not consider Puma at all. But that's definitively something we have to think about next.